### PR TITLE
Match pod `dnsPolicy` to `hostNetwork` config

### PR DESCRIPTION
**Pull request by gai6948**
Resolves https://github.com/open-telemetry/opentelemetry-operator/issues/690

I added a simple check to make the pods' DNSPolicy matching the `hostNetwork` option

### CODE REVIEW

File: pkg/collector/utils.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+package collector
+
+import (
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/open-telemetry/opentelemetry-operator/apis/v1alpha1"
+)
+
+func getDnsPolicy(otelcol v1alpha1.OpenTelemetryCollector) corev1.DNSPolicy {
+	dnsPolicy := corev1.DNSClusterFirst
```
Comment (pavolloffay) [inline pkg/collector/utils.go:24]: Shouldn't this be rather "Default" ? 

https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-dns-policy

**Review by pavolloffay — COMMENTED** [review]
Please add test to validate this.

File: pkg/collector/utils.go (hunk 1 above)
Comment (gai6948) [reply pkg/collector/utils.go:24]: Found this section on that page:
```
Note: "Default" is not the default DNS policy. If dnsPolicy is not explicitly specified, then "ClusterFirst" is used.
```

**Comment by gai6948** [comment]
Actually I don't know if `hostNetwork=true` makes sense for deployment and statefulsets, if it isn't perhaps we only need to test on daemonsets and not needing to duplicate such simple tests

**Comment by pavolloffay** [comment]
> Actually I don't know if hostNetwork=true makes sense for deployment and statefulsets, if it isn't perhaps we only need to test on daemonsets and not needing to duplicate such simple tests

This is why we should understand why people want to use `hostNetwork`

**Comment by pavolloffay** [comment]
@gai6948 the CI is still failing? Please fix it to move this forward.

**Comment by gai6948** [comment]
The tests should pass now

**Review by pavolloffay — APPROVED** [review]
